src/lib.rs

```diff
@@ -90,9 +88,13 @@ macro_rules! lsp_enum {
         impl std::convert::TryFrom<&str> for $typ {
             type Error = &'static str;
             fn try_from(value: &str) -> Result<Self, Self::Error> {
-                match value {
+                match () {
                     $(
-                        _ if { let (buf, len)  = crate::fmt_pascal_case_const(stringify!($name)); &buf[..len] == value.as_bytes() } => Ok(Self::$name),
+                        _ if {
+                            const X: (crate::PascalCaseBuf, usize) = crate::fmt_pascal_case_const(stringify!($name));
+                            let (buf, len) = X;
+                            &buf[..len] == value.as_bytes()
```

Comment:
pretty nice - this commit reduces the instruction count of the `CompletionItemKind::try_from()` from 4000 to just above 300 (in release mode, debug has around 10x more). Now let's hope that future rust will do this without the explicit `const` :)